### PR TITLE
fix: temporarily limit the size of the response until memory leak is resolved

### PR DESCRIPTION
**Pull request by alanshaw**
This should allow us to get freeway back into production while we fix the memory leak.

### CODE REVIEW

File: src/middleware.js [hunk 1]
```diff
@@ -155,3 +155,27 @@ export function withVersionHeader (handler) {
...
+ */
+export function withMaxContentLength (maxLength, handler) {
+  /**
+   * @type {import('@web3-storage/gateway-lib').Handler<import('@web3-storage/gateway-lib').Context>}
+   */
+  return async (request, env, ctx) => {
+    const response = await handler(request, env, ctx)
+    if (!response.headers.has('Content-Length')) return response
+
+    const contentLength = parseInt(response.headers.get('Content-Length') || '0')
```
Comment (olizilla) [inline src/middleware.js:174]: With the default value being provided, we can just collapse this to
```suggestion
    const contentLength = parseInt(response.headers.get('Content-Length') ?? '0')
```

**Review by olizilla — APPROVED** [review]
LGTM. When the tests say it works it's good to go.

**Review by vasco-santos — APPROVED** [review]
LGTM

File: src/middleware.js (hunk 1 above)
Comment (alanshaw) [reply src/middleware.js:174]: Hmm, I thought it returned `null` and `??` only work with `undefined`?

Comment (alanshaw) [reply src/middleware.js:174]: oh it's nullish...

Comment (alanshaw) [reply src/middleware.js]: 👍 thanks!

**Pull request merged**